examples/multisig_transfer.ts

```diff
@@ -93,11 +93,39 @@ function createOrApproveMultisigProposal<
       other_signatories: signatories.filter((value) => value !== pair.publicKey),
       store_call: false,
       max_weight: {
-        ref_time: 500_000_000n,
+        ref_time: 0n,
         proof_size: 0n,
       },
       maybe_timepoint: maybeTimepoint as Rest[0],
     }),
+  })
+    .feeEstimate
+    .access("weight")
+    .next((weight) => {
+      return {
+        ref_time: BigInt(weight.ref_time),
+        proof_size: BigInt(weight.proof_size),
+      }
+    })
```

Comment:
@tjjfvi Would you know if this is codec related?

`ref_time` and `proof_size` are `u64` but they are not decoded as biging

See https://github.com/paritytech/substrate/blob/59da38b3a07b4ddf380ff9b01a5874fd883138a9/primitives/weights/src/weight_v2.rs#L29-L35

Reply:
you are right

It might be a good DX to have the codec convert `number` to `bigint` when it's needed, WDYT?

Reply:
Having any implicit coercion legitimizes implicit coercion in other cases, which could quickly get out of hand (c.f. `@polkadot/types`, which is a rat's nest of implicit coercion).

Additionally, all codecs from `scale-ts` currently roundtrip exactly for every valid input (that is,
```ts
!$.is($foo, value) || deepEqual(value, $foo.decode($foo.encode(value)))
```
is always true), and I'd like to preserve this behavior.

Reply:
I think it would probably make the most sense for the `.feeEstimate` effect to do this BigInt coercion.

Reply:
if we do it in the effect, as soon as the `weight` shape changes, the effect will need to be refactored (like it happened with recent polkadot updates).

that's why I believe that implicit coercion from `number` to `bigint` will scale better for this case.